internal/impl/babysitter.go

```diff
@@ -170,14 +182,29 @@ func (b *babysitter) watchPods(ctx context.Context, component string) error {
 	b.watching[component] = struct{}{}
 	b.mu.Unlock()
 
+	// clear b.watching before leaving
+	defer func() {
+		b.mu.Lock()
+		delete(b.watching, component)
+		b.mu.Unlock()
+	}()
+
```

Comment:
after this point, returning errors without cleaning is dangerous.... since we wouldn't be able to watch this component again.